### PR TITLE
A selection of new Lua packages

### PR DESCRIPTION
**Pull request by maxguru**

### CODE REVIEW

File: lang/cgilua/Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  URL:=https://github.com/keplerproject/cgilua
+  DEPENDS:=+lua +luafilesystem
+endef
+
+define Package/cgilua/description
+	CGILua is a tool for creating dynamic HTML pages and manipulating input data from Web forms.
+endef
+
+define Build/Configure
+endef
```
Comment (karlp) [inline lang/cgilua/Makefile]: don't need this block.

Comment (4L3Xx) [inline lang/cgilua/Makefile]: Can you please clarify?  Why?  Which block exactly?  I'm pretty sure define Package/cgilua is needed.  Maybe define Build/Configure?

Comment (karlp) [inline lang/cgilua/Makefile]: Yes, the configure blocks. (in all these packages)

Comment (maxguru) [inline lang/cgilua/Makefile]: Please take a look at my fix.

**Comment by diizzyy** [comment]
@karlp 
Can you please have a look at this?

File: lang/cgilua/Makefile [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2016 OpenWrt.org
```
Comment (karlp) [inline lang/cgilua/Makefile]: This is not correct. You aren't OpenWrt.  Either _drop_ teh copyright outright (my choice) or provide your own name here. (will apply in all places)

File: lang/cgilua/Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/keplerproject/cgilua/archive/
+PKG_MD5SUM:=ed162ca341d0f1cd16868c80bed3da4c
+
+PKG_MAINTAINER:=Alex R <alex@totalwebservices.net>
+PKG_LICENSE=MIT
+
+LUA_MODULE_PATH:=/usr/lib/lua
+LUA_BIN_PATH:=/usr/bin
```
Comment (karlp) [inline lang/cgilua/Makefile]: Why are these two lines needed?

File: lang/cgilua/Makefile [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=cgilua
+PKG_VERSION:=5.1.4
+PKG_RELEASE:=1
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline lang/cgilua/Makefile]: This is a no go, use the git approach instead including xz tarballs.

File: lang/cgilua/Makefile [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	CGILua is a tool for creating dynamic HTML pages and manipulating input data from Web forms.
+endef
+
+define Build/Compile
+endef
+
+define Package/cgilua/install
+	$(INSTALL_DIR) $(1)/$(LUA_MODULE_PATH)/cgilua
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/src/cgilua/cgilua.lua $(1)/$(LUA_MODULE_PATH)/cgilua.lua
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/src/cgilua/authentication.lua $(1)/$(LUA_MODULE_PATH)/cgilua/authentication.lua
```
Comment (karlp) [inline lang/cgilua/Makefile]: Really? the upstream package keeps them in the same dir and needs them in different dirs?  Does the upstream package have a makefile with an "install" target you can use?

File: lang/jsonrpc4lua/Makefile [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#  
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=jsonrpc4lua
+PKG_VERSION:=1.0.1
+PKG_RELEASE:=2
```
Comment (karlp) [inline lang/jsonrpc4lua/Makefile]: no. pkg_release is obviously 1 here.

File: lang/jsonrpc4lua/Makefile [hunk 7]
```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (C) 2006-2015 OpenWrt.org
```
Comment (karlp) [inline lang/jsonrpc4lua/Makefile]: definitely not 2006! (or 2015!)

File: lang/jsonrpc4lua/Makefile [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  SUBMENU:=Lua
+  SECTION:=lang
+  CATEGORY:=Languages
+  TITLE:=jsonrpc4lua
+  URL:=https://github.com/pdxmeshnet/jsonrpc4lua
+  DEPENDS:=+lua +luasocket +lua-cjson
+endef
+
+define Package/jsonrpc4lua/description
+	JSONRPC for Lua
```
Comment (karlp) [inline lang/jsonrpc4lua/Makefile]: Great description!  Is this for clients? for servers? for generating stubs only? Who knows!

File: lang/lua-bit32/Makefile [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+   bit32 is the native Lua 5.2 bit manipulation library, in the version from Lua 5.3; it is compatible with Lua 5.1, 5.2 and 5.3.
+endef
+
+TARGET_CFLAGS += $(FPIC) -shared
+TARGET_LDFLAGS += -llua
+
+LUA_MODULE_PATH:=/usr/lib/lua
+
+define Build/Compile
+	$(TARGET_CROSS)gcc $(TARGET_CFLAGS) $(TARGET_LDFLAGS) -I $(PKG_BUILD_DIR)/c-api/ -o $(PKG_BUILD_DIR)/bit32.so $(PKG_BUILD_DIR)/lbitlib.c
```
Comment (karlp) [inline lang/lua-bit32/Makefile]: Is upstream _really_ that broken that you have to compile it yourself? Have you considered just using lua-bitop? it's already packaged.

File: lang/lua-inifile/Makefile [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_URL:=git://github.com/bartbes/love-misc-libs.git
+PKG_SOURCE_VERSION:=master
+
+PKG_MAINTAINER:=Alex R <alex@totalwebservices.net>
+PKG_LICENSE=Simplified BSD license
```
Comment (karlp) [inline lang/lua-inifile/Makefile]: https://spdx.org/licenses/  I think you want "BSD-2-Clause" here

File: lang/luaproc/Makefile [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=luaproc
+PKG_VERSION:=1.0
+PKG_RELEASE:=4
```
Comment (karlp) [inline lang/luaproc/Makefile]: no, should be 1

**Comment by karlp** [comment]
You've used different names and emails for some of these packages, was that intentional?  Are you really sure that none of the existing packages met your needs for any of these?

File: lang/cgilua/Makefile (hunk 2 above)
Comment (maxguru) [inline lang/cgilua/Makefile]: I was under the impression that was required.  I'll fix it.

Comment (maxguru) [inline lang/cgilua/Makefile]: Actually, according to [this](https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md) I *can* use OpenWrt.

File: lang/cgilua/Makefile (hunk 4 above)
Comment (maxguru) [inline lang/cgilua/Makefile]: Why not? What rule does this break?

Comment (diizzyy) [inline lang/cgilua/Makefile]: You'll run into file names clashing if they having nothing more than v<version>.tar.gz as file name.

**Comment by maxguru** [comment]
Regarding different names/emails, I was still in the process of separating identities.

Regarding possible duplication of functionality, I don't believe that's a valid concern.

jsonrpc4lua: The original json4lua package was only semi-functional.  The JSON-RPC server component didn't work at all.  Initially, I was trying to fix it (see https://github.com/craigmj/json4lua/pull/13) but then, I realized it was trying to do too much.  JSON-RPC features should not be bundled into JSON encoding/decoding library.  The new fork uses lua-cjson for encoding/decoding and has *actually functioning* JSON-RPC client and server code.

bit32:  I was surprised this wasn't already in the repository, given the version of Lua that was used. "bit32 is the native Lua 5.2 bit manipulation library, backported to Lua 5.1"

alt-getopt:  I found this package useful, but it can be left out.
inifile: I found this package useful, but it can be left out.

cgilua: I was also surprised that the repository is missing this package, given that it has lua-wsapi and lua-xavante.  Also, I believe this is a dependency for the server component of jsonrpc4lua (and the non-functioning JSON-RPC server code in json4lua).

File: lang/cgilua/Makefile (hunk 2 above)
Comment (diizzyy) [inline lang/cgilua/Makefile]: It's incorrect unfortunately. Either drop it as karlp suggests or add your own.

**Comment by maxguru** [comment]
I forgot to talk about luaproc. It is an excellent library.  Very good alternative to Lua Lanes.

File: lang/cgilua/Makefile (hunk 4 above)
Comment (maxguru) [inline lang/cgilua/Makefile]: Oh, right.

File: lang/lua-bit32/Makefile (hunk 9 above)
Comment (maxguru) [inline lang/lua-bit32/Makefile]: If I recall, at the time bit32 looked better and more future proof.  I didn't realize there were other ways to compile it.  I'll have to check if I can rework it.

File: lang/cgilua/Makefile (hunk 5 above)
Comment (maxguru) [inline lang/cgilua/Makefile]: I'll have to test to make sure the other way works.

**Comment by karlp** [comment]
You'll want to squish the fixup commits at some point too btw.

**Comment by maxguru** [comment]
Rant:

Here is the problem.  I was working on the project that triggered the commits prior to June of 2016.  I created the pull request because I thought my packages were useful.  Nobody responded to my pull request back then.  Then after all these months somebody decided to respond finally.  I already moved on and no longer have the environment set up.  Once I get back to the project, I will see about fixing the couple of issues that I still have left to fix.  If somebody else is up for it before then, feel free to do it.

@champtar, I wasn't aware that I needed to sign off commits.  I would need to recreate the pull request to fix this, no?  I might as well create individual pull requests for each package if that is the case.

**Comment by diizzyy** [comment]
@karlp  @thess @hnyman 
Close due to submitter timeout?

**Comment by thess** [comment]
Way to old - closing.

**Pull request closed without merging**